src/managementgroup/azext_managementgroup/generated/_params.py

```diff
@@ -0,0 +1,166 @@
+        c.argument('display_name', type=str, help='The friendly name of the management group. If no value is passed '
+                   'then this  field will be set to the groupId.')
+        c.argument('id_', options_list=['--id'], type=str, help='The fully qualified ID for the parent management '
+                   'group.  For example, /providers/Microsoft.Management/managementGroups/0000000-0000-0000-0000-000000'
+                   '000000', arg_group='Details Parent')
+
+    with self.argument_context('managementgroup management-group update') as c:
+        c.argument('group_id', type=str, help='Management Group ID.')
+        c.argument('display_name', type=str, help='The friendly name of the management group.')
+        c.argument('parent_group_id', type=str, help='(Optional) The fully qualified ID for the parent management '
+                   'group.  For example, /providers/Microsoft.Management/managementGroups/0000000-0000-0000-0000-000000'
+                   '000000')
+
+    with self.argument_context('managementgroup management-group delete') as c:
+        c.argument('group_id', type=str, help='Management Group ID.')
+
+    with self.argument_context('managementgroup management-group show-descendant') as c:
+        c.argument('group_id', type=str, help='Management Group ID.')
+        c.argument('skiptoken', type=str, help='Page continuation token is only used if a previous operation returned '
+                   'a partial result.  If a previous response contains a nextLink element, the value of the nextLink '
+                   'element will include a token parameter that specifies a starting point to use for subsequent '
+                   'calls.')
+        c.argument('top', type=int, help='Number of elements to return when retrieving results. Passing this in will '
+                   'override $skipToken.')
+
+    with self.argument_context('managementgroup management-group wait') as c:
+        c.argument('group_id', type=str, help='Management Group ID.')
+        c.argument('expand', arg_type=get_enum_type(['children', 'path', 'ancestors']), help='The $expand=children '
+                   'query string parameter allows clients to request inclusion of children in the response payload.  '
+                   '$expand=path includes the path from the root group to the current group.  $expand=ancestors '
+                   'includes the ancestor Ids of the current group.')
+        c.argument('recurse', arg_type=get_three_state_flag(), help='The $recurse=true query string parameter allows '
+                   'clients to request inclusion of entire hierarchy in the response payload. Note that  '
+                   '$expand=children must be passed up if $recurse is set to true.')
+        c.argument('filter_', options_list=['--filter'], type=str, help='A filter which allows the exclusion of '
+                   'subscriptions from results (i.e. \'$filter=children.childType ne Subscription\')')
```

Comment:
Same as above. In addition, please abstract the repeatedly used parameters into `CLIArgumentType`, so that they can be reused in other places